app/scripts/directives/labels.js

```diff
@@ -50,6 +66,32 @@ angular.module('openshiftConsole')
         if (!angular.isDefined(attrs.canToggle)) {
           scope.canToggle = true;
         }
-      }
+      },
+      controller: [
+        '$scope',
+        function($scope) {
+
```

Comment:
This is a port of the label validation logic from osc-key-values.  The "shared functions" are up above.  This object is a "pseudo regex" with a `.test()` method because its easier to pass to `key-value-editor`, which is using `ng-pattern` and expects a regex or "regex like" object. I went this route instead of passing a function due to wanting the error messages to still key off of `form.input.$error.pattern` w/o having to handle that manually.